diff --git a/node_modules/react-native-root-toast/index.d.ts b/node_modules/react-native-root-toast/index.d.ts
index 88c2647..0b8f453 100644
--- a/node_modules/react-native-root-toast/index.d.ts
+++ b/node_modules/react-native-root-toast/index.d.ts
@@ -41,7 +41,7 @@ declare module "react-native-root-toast"{
         CENTER:number,
     }
     export default class Toast extends React.Component<ToastProps>{
-        static show:(message:string,options?:ToastOptions)=>any;
+        static show:(message:string | React.ReactNode,options?:ToastOptions)=>any;
         static hide:(toast:any)=>void;
         static durations:Durations;
         static positions:Positions;
diff --git a/node_modules/react-native-root-toast/lib/ToastContainer.js b/node_modules/react-native-root-toast/lib/ToastContainer.js
index a645d4d..e1c7686 100644
--- a/node_modules/react-native-root-toast/lib/ToastContainer.js
+++ b/node_modules/react-native-root-toast/lib/ToastContainer.js
@@ -255,13 +255,20 @@ class ToastContainer extends Component {
                     pointerEvents="none"
                     ref={ele => this._root = ele}
                 >
-                    <Text style={[
-                        styles.textStyle,
-                        props.textStyle,
-                        props.textColor && {color: props.textColor}
-                    ]}>
-                        {this.props.children}
-                    </Text>
+                    {typeof this.props.children === 'string' ?
+                        <Text style={[
+                            styles.textStyle,
+                            props.textStyle,
+                            props.textColor && {color: props.textColor}
+                        ]}>
+                            {this.props.children}
+                        </Text>
+                        :
+                        <View>
+                            {this.props.children}
+                        </View>
+                    }
+                    
                 </Animated.View>
             </TouchableWithoutFeedback>
         </View> : null;
